src/test/test_pidfile.sh

```diff
@@ -61,15 +61,41 @@ function teardown_unexist_pidfile() {
     return $status
 }
 
-function TEST_contend_pidfile() {
+function TEST_pidfile() {
     local dir=$1
     setup $dir 
- 
-    run_mon $dir a 
+
+    # no daemon can use a pidfile that is owned by another daemon
+    run_mon $dir a || return 1
     run_mon $dir a 2>&1 | grep "failed to lock pidfile" || return 1
 
-    run_osd $dir 0 
+    run_osd $dir 0 || return 1
     run_osd $dir 0 2>&1 | grep "failed to lock pidfile" || return 1
+
+    # when a daemon shutdown, it will not unlink a path different from
+    # the one it owns
+    mv $dir/osd.0.pid $dir/osd.0.pid.old || return 1
+    cp $dir/osd.0.pid.old $dir/osd.0.pid || return 1
+    kill_daemons $dir TERM osd.0 || return 1
```

Comment:
thought it would not work, but seems `kill_daemon` is able to match `osd.0.pid.old`, so all good!